charmcraft/commands/analyze.py

```diff
@@ -75,6 +83,20 @@ def run(self, parsed_args):
             override_ignore_config=override_ignore_config,
         )
 
+        # if format is json almost no further processing is needed
+        if parsed_args.format == JSON_FORMAT:
+            info = [
+                {
+                    "name": r.name,
+                    "result": r.result,
+                    "url": r.url,
+                    "type": r.check_type,
+                }
+                for r in linting_results
+            ]
+            logger.info(json.dumps(info))
```

Comment:
wouldn't we preffer to pretty print the output? `indent=4` to `json.dumps`

Reply:
Ooooooh, very nice!! Used it!